highlight/src/c_lib.rs

```diff
@@ -145,42 +152,60 @@ pub extern "C" fn ts_highlight_buffer_new() -> *mut TSHighlightBuffer {
     }))
 }
 
+/// Deleteis a [`TSHighlighter`] instance.
+///
+/// # Safety
+///
+/// `this` must be non-null.
```

Comment:
> `this` must be non-null.

This is not enough guaranty for safety, it also should be a valid pointer to a valid structure.